src/core/src/op/sign.cpp

```diff
@@ -44,7 +44,7 @@ bool Sign::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
 
     using namespace ov::element;
     return IF_TYPE_OF(v0_Sign_evaluate,
-                      OV_PP_ET_LIST(f16, f32, i32, i64, u32, u64),
+                      OV_PP_ET_LIST(bf16, f16, f32, f64, i4, i8, i16, i32, i64, u1, u4, u8, u16, u32, u64),
```

Comment:
test case for reference implementation changes added.
I think currently there is no particular reason/request to extend evaluate types of Sign op, I previously tried to extend it just because when implement this PR, I found its types are not match with template plugin type list. But considering the binary size as you mentioned, it seems reasonable to keep the most common types. Have changed back.